File: BitcoinCheckerDataModule/src/com/mobnetic/coinguardian/util/TimeUtils.java
package com.mobnetic.coinguardian.util;

public class TimeUtils {

	public final static long NANOS_IN_MILLIS = 1000;
	public final static long MILLIS_IN_SECOND = 1000;
	public final static long MILLIS_IN_MINUTE = 60*MILLIS_IN_SECOND;
	public final static long MILLIS_IN_HOUR = 60*MILLIS_IN_MINUTE;
	public final static long MILLIS_IN_DAY = 24*MILLIS_IN_HOUR;
	private final static long MILLIS_IN_YEAR = 365*MILLIS_IN_DAY;
	
	public static long parseTimeToMillis(long time) {
		if(time<MILLIS_IN_YEAR)
			return time*MILLIS_IN_SECOND;
		else if(time>5000*MILLIS_IN_YEAR)
			return time/1000;
		return time;
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/util/TimeUtils.kt
package com.aneonex.bitcoinchecker.datamodule.util

import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter

object TimeUtils {
//    const val NANOS_IN_MILLIS: Long = 1000
    const val MILLIS_IN_SECOND: Long = 1000
    const val MILLIS_IN_MINUTE = 60 * MILLIS_IN_SECOND
    const val MILLIS_IN_HOUR = 60 * MILLIS_IN_MINUTE
    const val MILLIS_IN_DAY = 24 * MILLIS_IN_HOUR
    const val MILLIS_IN_YEAR = 365 * MILLIS_IN_DAY

    fun parseTimeToMillis(time: Long): Long {
        if (time < MILLIS_IN_YEAR)
            return time * MILLIS_IN_SECOND
        else
            if (time > 5000 * MILLIS_IN_YEAR)
                return time / 1000

        return time
    }

    // Parsing string and converting to timestamp (in milliseconds).
    //  Returns 0 if parsing failed
    fun convertISODateToTimestamp(isoDateString: String): Long {
        return ZonedDateTime.parse(isoDateString, DateTimeFormatter.ISO_DATE_TIME).toEpochSecond() * MILLIS_IN_SECOND
    }
}